src/darwin/Framework/CHIPTests/MTRPerControllerStorageTests.m

```diff
@@ -2298,4 +2298,496 @@ - (void)testSubscriptionPool
     [self doTestSubscriptionPoolWithSize:2 deviceOnboardingPayloads:deviceOnboardingPayloads];
 }
 
+- (MTRDevice *)getMTRDevice:(NSNumber *)deviceID
+{
+    __auto_type * factory = [MTRDeviceControllerFactory sharedInstance];
+    XCTAssertNotNil(factory);
+
+    __auto_type * rootKeys = [[MTRTestKeys alloc] init];
+    XCTAssertNotNil(rootKeys);
+
+    __auto_type * operationalKeys = [[MTRTestKeys alloc] init];
+    XCTAssertNotNil(operationalKeys);
+
+    NSNumber * nodeID = @(123);
+    NSNumber * fabricID = @(456);
+
+    NSError * error;
+    __auto_type * storageDelegate = [[MTRTestPerControllerStorageWithBulkReadWrite alloc] initWithControllerID:[NSUUID UUID]];
+    MTRPerControllerStorageTestsCertificateIssuer * certificateIssuer;
+    MTRDeviceController * controller = [self startControllerWithRootKeys:rootKeys
+                                                         operationalKeys:operationalKeys
+                                                                fabricID:fabricID
+                                                                  nodeID:nodeID
+                                                                 storage:storageDelegate
+                                                                   error:&error
+                                                       certificateIssuer:&certificateIssuer];
+    XCTAssertNil(error);
+    XCTAssertNotNil(controller);
+    XCTAssertTrue([controller isRunning]);
+
+    XCTAssertEqualObjects(controller.controllerNodeID, nodeID);
+
+    certificateIssuer.nextNodeID = deviceID;
+    [self commissionWithController:controller newNodeID:deviceID];
+
+    MTRDevice * device = [MTRDevice deviceWithNodeID:deviceID controller:controller];
+    return device;
+}
+
+- (NSMutableArray<NSNumber *> *)getEndpointArrayFromPartsList:(NSDictionary *)partsList forDevice:(MTRDevice *)device
+{
+    // Initialize the endpoint array with endpoint 0.
+    NSMutableArray<NSNumber *> * endpoints = [NSMutableArray arrayWithObject:@0];
+
+    [endpoints addObjectsFromArray:[device arrayOfNumbersFromAttributeValue:[device _dataValueWithoutDataVersion:partsList]]];
```

Comment:
Why does this need `_dataValueWithoutDataVersion`?

Reply:
not needed. removed